cloudbuild/periodic/run-all-tests.sh

```diff
@@ -0,0 +1,108 @@
+#!/bin/bash
+
+set -uxo pipefail
+
+COMMIT=$(git rev-parse HEAD)
+TESTS_TO_RUN=(
+  ":DataprocInitActionsTestSuite"
+) 
+
+run_tests() {
+  bazel test \
+  	--jobs=15 \
+  	--local_cpu_resources=15 \
+  	--local_ram_resources=$((15 * 1024)) \
+  	--action_env=INTERNAL_IP_SSH=true \
+  	--test_output=errors \
+  	--noshow_progress \
+  	--noshow_loading_progress \
+  	--test_arg="--image_version=${IMAGE_VERSION}" \
+  	"${TESTS_TO_RUN[@]}"
+}
+
+get_build_num() {
+	build_num=$(($(gsutil cat gs://init-actions-github-tests/counter.txt)-1))
+	echo $build_num
+}
+
+BUILD_NUM=$(get_build_num)
+
+# Reads the test logs of a given component by compiling all the logs of all its shards
+get_test_logs() {
+	component=$1
+  SHARD_PATHS=(bazel-testlogs/${component}/test_${component}/shard*)
+  #SHARD_DIRS=("${SHARD_PATHS[@]##*/}")
+  all_logs=""
+  for shard in "${SHARD_PATHS[@]}"; do
+    logs_filepath=$(readlink -f ${shard}/test.log)
+    logs=$(cat $logs_filepath)
+    all_logs="${all_logs} \n\n ${logs}"
+  done
+	echo -e $all_logs
+}
+
+get_test_status() {
+	component=$1
+  SHARD_PATHS=(bazel-testlogs/${component}/test_${component}/shard*)
+  #SHARD_DIRS=("${SHARD_PATHS[@]##*/}")
+  declare -a test_results=()
+  for shard in "${SHARD_PATHS[@]}"; do
+  	xml_filepath=$(readlink -f ${shard}/test.xml)
+    # Parse the XML and determine if test passed or failed
+    failures=$(xmllint --xpath 'string(/testsuites/@failures)' ${xml_filepath})
+    errors=$(xmllint --xpath 'string(/testsuites/@errors)' ${xml_filepath})
+    if [ "$errors" == "0" ] && [ "$failures" == "0" ]; then
+      test_results+=("SUCCESS")
+    else
+      test_results+=("FAILURE")
+    fi
+  done
+  # If there's a single test shard that fails, the test is considered a failure.
+  if [[ " ${test_results[@]} " =~ "FAILURE" ]]; then
+    status="FAILURE"
+  else
+    status="SUCCESS"
+  fi
+	echo $status
+}
+
+create_finished_json() {
+  component=$1
+  status=$(get_test_status $component)
+  jq -n \
```

Comment:
Why do you need `jq` for this? seems like you can create JSON string without it pretty easily.

Reply:
I like that jq ensures that the created file has a correct JSON format. Doing it by creating the string also works and is simple but with jq it's more reliable because whatever is uploaded to GCS will always be a valid JSON file that is readable. For example, doing it by string creation, a typo will not fail the file creation process causing a wrong JSON file to be uploaded. I'd rather have it fail when it's trying to create it.